core/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -34,12 +34,20 @@
 
 import java.io.IOException;
 import java.util.Map;
+import java.util.function.Supplier;
 
 /**
  * Script holds all the parameters necessary to compile or find in cache and then execute a script.
  */
 public class Script implements ToXContent, Streamable {
 
+    public static final Supplier<Script> SUPPLIER = new Supplier<Script>() {
```

Comment:
can you add javadocs that this is for usage in conjunction with StreamInput.readStreamable?